src/T2DMap.cpp

```diff
@@ -2351,7 +2351,7 @@ void T2DMap::paintMapInfo(const QElapsedTimer& renderTimer, QPainter& painter, c
 
 void T2DMap::mouseDoubleClickEvent(QMouseEvent* event)
 {
-    if (mDialogLock) {
+    if (mDialogLock || (event->buttons() & Qt::RightButton)) {
```

Comment:
Not sure what you are trying to achieve here - however I have just detected a bug that can be cured here - I have a mouse with more than three buttons and all but the right button are activating this code (and thus the speed walk code. Note that `event->buttons()` for pressed and doubleClicked events is **all** the buttons that are pressed when this event is generated *including* the one that initiated the event - whereas `event->button()` is the one that *caused* the event. So perhaps we need to use:
```suggestion
    if (mDialogLock || (event->buttons() != Qt::LeftButton)) {
```
which will abort the following code if anything other than the *only* button being down is the left one...